lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -1,45 +1,106 @@
+import 'dart:async';
+
 import 'package:pokedex_start/api/model/pokemon.dart';
-import 'package:pokedex_start/features/pokemon_overview/widget/pokemon_card.dart';
+import 'package:pokedex_start/features/pokemon_overview/widget/pokemon_grid_view.dart';
+import 'package:pokedex_start/features/pokemon_overview/widget/search_bar.dart';
 import 'package:pokedex_start/utilities/async.dart';
-import 'package:pokedex_start/utilities/constant.dart';
 import 'package:pokedex_start/utilities/string_constant.dart';
+import 'package:pokedex_start/widget/app_text.dart';
+import 'package:pokedex_start/widget/spacing.dart';
 import 'package:flutter/material.dart';
 
-class PokemonOverviewPage extends StatelessWidget {
+class PokemonOverviewPage extends StatefulWidget {
   const PokemonOverviewPage({
     required this.pokemons,
+    required this.searchedPokemons,
+    required this.filterPokemons,
+    required this.clearSearchedPokemons,
     super.key,
   });
 
   final Async<List<Pokemon>> pokemons;
+  final List<Pokemon> searchedPokemons;
+  final VoidCallback clearSearchedPokemons;
+  final Function(String) filterPokemons;
```

Comment:
also maybe rename this to onSearchPokemons, same goes with the functions declared in your vm 🙂